### DISCUSSION TITLE
ReadBinaryFile error

### DISCUSSION THREAD

**Discussion by JebinBabu** [neutralinojs/neutralinojs#866] [Q&A] [answered]
{code: 'NE_RT_NATRTER', message: "Native method execution error occurred. Failed bec…hat you've provided required parameters properly."}

This is the error showing up when using readBinaryFile

I created a binary file using writeBinaryFile method and tried to read the created file using readBinaryFile method. But it seems there is some problem. I tried it on other files also, but the same error appears.

**Comment by pegvin** [comment]
Can you share the code too? This error occurs because of:
> Native method runtime error. Mostly occured due to missing parameters.

**Reply by JebinBabu** [reply]
let data = await Neutralino.filesystem.readBinaryFile({
  fileName: './myFile.bin'
});
let view = new Uint8Array(data);

console.log('Binary content: ', view);

I just copied and pasted the same code from the Docs.

let rawBin = new ArrayBuffer(1);
let view = new Uint8Array(rawBin);
view[0] = 64; // Saves ASCII '@' to the binary file

await Neutralino.filesystem.writeBinaryFile('./myFile.bin', rawBin);

also created the bin file from the code snippet from the Docs

**Comment by pathange-s** [comment, marked as answer]
Try this instead. The error occurred because of the incorrect passing of parameters to the function.

```
      let rawBin = new ArrayBuffer(1);
      let view = new Uint8Array(rawBin);      
      view[0] = 64; // Saves ASCII '@' to the binary file
      let data = await Neutralino.filesystem.readBinaryFile(NL_PATH + './myFile.bin');
      view = new Uint8Array(data);
      console.log('Binary content: ', view);
```

Not sure why, documentation might be wrong. But in the main framework, `json readBinaryFile(const json &input);` this is what it is defined as.

**Reply by pathange-s** [reply]
Let me know if this resolves your issue.

**Reply by JebinBabu** [reply]
now it works perfectly. Thanks

**Reply by pathange-s** [reply]
Pleasure 🤗 !
